Content.Server/Atmos/Piping/Unary/EntitySystems/GasVentScrubberSystem.cs

```diff
@@ -59,40 +56,35 @@ private void OnVentScrubberUpdated(EntityUid uid, GasVentScrubberComponent scrub
             || !EntityManager.TryGetComponent(uid, out NodeContainerComponent? nodeContainer)
             || !nodeContainer.TryGetNode(scrubber.OutletName, out PipeNode? outlet))
             {
-                appearance?.SetData(ScrubberVisuals.State, ScrubberState.Off);
-                _ambientSoundSystem.SetAmbience(scrubber.Owner, false);
                 return;
             }
-            _ambientSoundSystem.SetAmbience(scrubber.Owner, true);
 
-            var environment = _atmosphereSystem.GetTileMixture(EntityManager.GetComponent<TransformComponent>(scrubber.Owner).Coordinates, true);
+            var xform = Transform(uid);
+            var environment = _atmosphereSystem.GetTileMixture(xform.Coordinates, true);
```

Comment:
Both this and the _atmosphereSystem.GetAdjacentTileMixture need to convert entity-coordinates into a grid & grid indices/transforms, and then get the `GridAtmosphereComponent`. It would be slightly faster to call use the overrides of this function that directly take in a `GridAtmosphereComponent` and grid indices. 

Pretty minor, but as this happens for ever scrubber/vent on the station it might add up.

Reply:
ah never mind this is only for wide-net mode, which usually isn't on. So yeah should just get fixed whenever the data is cached.

Reply:
yeah i looked into passing the tile mix through the update event for anchored entities but it'd be a bit of an undertaking